src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -310,5 +312,63 @@ public IVsPathContext GetSolutionPathContext()
                 pathContext.FallbackPackageFolders.Cast<string>(),
                 trie);
         }
+
+        private IList<Project> GetProjectsInSolution(DTE dte)
+        {
+            var supportedProjects = dte.Solution.Projects;
+            Projects projects = dte.Solution.Projects;
+
+            List<Project> projectList = new List<Project>();
+            var item = projects.GetEnumerator();
+            while (item.MoveNext())
```

Comment:
Can't you just use a `foreach` statement instead of trying to implement `foreach` yourself?
```C#
foreach (Project project in projects)
{
    …
}
```

Reply:
I copied this from other VS repos, didn't want to change anything. But you're right, foreach seems safe and logical.